src/models/MyAccount.ts

```diff
@@ -106,85 +113,111 @@ export default class MyAccount {
   }
 
   buildHolds(holds: SierraHold[], bibData: SierraBibEntry[]): Hold[] {
-    const bibDataMap = MyAccount.buildBibData(bibData)
-    return holds.map((hold: SierraHold) => {
-      return {
-        patron: MyAccount.getRecordId(hold.patron),
-        id: MyAccount.getRecordId(hold.id),
-        pickupByDate: MyAccount.formatDate(hold.pickupByDate) || null,
-        canFreeze: hold.canFreeze,
-        frozen: hold.frozen,
-        status: MyAccount.getHoldStatus(hold.status),
-        pickupLocation: hold.pickupLocation,
-        title: bibDataMap[hold.record.bibIds[0]].title,
-        isResearch: bibDataMap[hold.record.bibIds[0]].isResearch,
-        bibId: hold.record.bibIds[0],
-        isNyplOwned: bibDataMap[hold.record.bibIds[0]].isNyplOwned,
-        catalogHref: bibDataMap[hold.record.bibIds[0]].isNyplOwned
-          ? bibDataMap[hold.record.bibIds[0]].isResearch
-            ? `https://nypl.org/research/research-catalog/bib/b${hold.record.bibIds[0]}`
-            : `https://nypl.na2.iiivega.com/search/card?recordId=${hold.record.bibIds[0]}`
-          : null,
-      }
-    })
+    let bibDataMap
+    try {
+      bibDataMap = MyAccount.buildBibData(bibData)
```

Comment:
`buildBibData` calls a reduce function but I think the default is an empty array. So here I don't expect `buildBibData` to throw an error, no?

Reply:
I'm worried about certain responses not having varFields. If you don't specify it in the url, you don't get varFields back.

Reply:
That's fair. If you throw an error, the entire page will crash. What if you can resolve a fallback for when responses don't have varFields so that the page can still render but not display that specific data?

Reply:
At the top level, `MyAccoutFactory` is called within a try/catch and there are defaults, or rather undefined values being returned. So taking a better look, this should crash. 

But what if all the fetch calls succeed but there's an error with the data manipulation once `new MyAccount` is called. In that case, will none of the data get returned?

Reply:
That's a good point. Even without this work, if there are any errors thrown here, then no data is returned.